org-roam-macs.el

```diff
@@ -64,6 +64,21 @@ to look.
   (when org-roam-verbose
     (apply #'message `(,(concat "(org-roam) " format-string) ,@args))))
 
+(defmacro org-roam--get-completions (query &rest processing)
+  "Return an alist for completion from QUERY and PROCESSING.
+QUERY is the string to be passed to `org-roam-db-query'.
+PROCESSING should contain the forms to be applied on every
+element matched by QUERY."
+  (declare (indent 1))
+  `(let* ((rows (org-roam-db-query ,query))
+          completions)
+     (seq-sort-by (lambda (x)
+                    (plist-get (nth 3 x) :mtime))
```

Comment:
this is dependent on meta always being the third column, seems brittle.

Reply:
Oh, hadn't noticed, that's true indeed.  Let me see if I can keep the macro useful whilst modularising the sorting as well.